catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/utils/FormulaEditorCategoryListDataInteractionWrapper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.ui.activity.utils;
+
+import android.support.test.espresso.DataInteraction;
+
+import org.catrobat.catroid.uiespresso.util.wrappers.DataInteractionWrapper;
+
+import static android.support.test.espresso.Espresso.onData;
+
+import static org.catrobat.catroid.uiespresso.util.matchers.FormulaEditorCategoryListMatchers.isFormulaEditorCategoryListView;
+import static org.catrobat.catroid.uiespresso.util.matchers.FormulaEditorCategoryListMatchers.withFormulaEditorCategoryItemName;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class FormulaEditorCategoryListDataInteractionWrapper extends DataInteractionWrapper {
+
+	public FormulaEditorCategoryListDataInteractionWrapper(DataInteraction dataInteraction) {
+		super(dataInteraction);
+	}
+
+	public static FormulaEditorCategoryListDataInteractionWrapper onFormulaEditorCategoryItemWithNameAtPosition(String formulaCategoryItem, int position) {
```

Comment:
i want to have the dataInteraction and viewInteraction wrappers to handle and feel the same as the erspresso internal ones, please have a look at BrickDataInteractionWrapper for example.

so to have every method having only one parameter, i dont want to combine the component and position in one method.
eg.

`public static FormulaEditorCategoryListDataInteractionWrapper onFormulaEditorCategory(String category) {
return new FormulaEditorCategoryListDataInteractionWrapper(onData(allOf(instanceOf(String.class), withFormulaEditorCategoryItemName(formulaCategoryItem)))
inAdapterView(isFormulaEditorCategoryListView());
}`

which then can be used as follows:
instead of 
`onFormulaEditorCategoryItemWithNameAtPosition("someCategory", 5)`
that would be 
`onFormulaEditorCategoryItemWithName("someCategory").atPosition(5)`

but then again, depending on if this is used in multiple test classes or not, it could be even put there as a function instead of separate class...